charts/aws-fsx-csi-driver/templates/controller-serviceaccount.yaml

```diff
@@ -1,22 +1,23 @@
-{{- if .Values.serviceAccount.create -}}
+{{- if .Values.controller.serviceAccount.create }}
 apiVersion: v1
 kind: ServiceAccount
 metadata:
-  name: {{ include "helm.serviceAccountName" . }}
+  name: {{ .Values.controller.serviceAccount.name }}
   labels:
-    {{- include "helm.labels" . | nindent 4 }}
-  {{- with .Values.serviceAccount.annotations }}
+    {{- include "aws-fsx-csi-driver.labels" . | nindent 4 }}
+  {{- with .Values.controller.serviceAccount.annotations }}
   annotations:
     {{- toYaml . | nindent 4 }}
   {{- end }}
+{{- end }}
```

Comment:
Moving this end here will create the ClusterRole and ClusterRoleBinding regardless of whether the service account is created.  I am not 100% sure what will happen with the ClusterRoleBinding since the service account it is referencing won't exist if `.Values.controller.serviceAccount.create` is false.  Seems like this end should move back to the end of the template.

Reply:
It's intentional to match how ebs does it https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/master/charts/aws-ebs-csi-driver/templates/serviceaccount-csi-controller.yaml (ebs splits up the serviceaccount from the associated RBAC bindings whereas here it's same fille...I kind of prefer keeping serviceaccount and associated RBAC in same file, but for ebs there are a LOT of RBAC things). The reasoning is that even if .Values.controller.serviceAccount.create is false, .Values.controller.serviceAccount.name is respected. An example use-case is on EKS if I take advantage of the "IAM for service accounts" feature then I will probably want to create the serviceAccount myself because I need it to have a special annotation. But I still need the ClusterRole and ClusterRoleBinding to exist for my serviceAccount whether it is created by me or by helm

Reply:
Personally, I would just add the annotation to `controller.serviceAccount.annotations`, which is what I did for the EBS chart.  I do, however, see what you are saying about some users potentially wanting to use a pre-existing service account.

If this chart isn't creating a service account and therefore expects that one already exists it might be worth using the lookup function to validate that it does exist and fail if it doesn't.
https://helm.sh/docs/chart_template_guide/functions_and_pipelines/#using-the-lookup-function